server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/QuotaMailingTest.scala

```diff
@@ -0,0 +1,234 @@
+    mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, ANDRE.asString, DefaultMailboxes.INBOX)
+
+    mailboxProbe.createMailbox(andreDraftsPath)
+
+    RestAssured.requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .build
+  }
+
+  @Category(Array(classOf[BasicFeature]))
+  @Test
+  def shouldSendANoticeWhenThresholdExceeded(server: GuiceJamesServer): Unit = {
+    val quotaProbe = server.getProbe(classOf[QuotaProbesImpl])
+    quotaProbe.setMaxStorage(quotaProbe.getQuotaRoot(MailboxPath.inbox(BOB)), QuotaSizeLimit.size(100 * 1000))
+
+    andreSendMailToBob(server)
+
+    // Bob receives a mail big enough to trigger a configured threshold
+    awaitAtMostTwoMinutes.until(() => listMessageIds().size == 2)
+
+    val ids: List[String] = listMessageIds().asScala.toList
+    val idString: String = concatMessageIds(ids)
+
+    val request =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core","urn:ietf:params:jmap:mail", "urn:ietf:params:jmap:submission"],
+         |  "methodCalls": [[
+         |     "Email/get",
+         |     {
+         |       "accountId": "$ACCOUNT_ID",
+         |       "ids": [$idString]
+         |     },
+         |     "c1"]]
+         |}""".stripMargin
+
+    `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post()
+    .`then`
+      .statusCode(SC_OK)
+      .log.ifValidationFails()
+      .body("methodResponses[0][1].list.subject",
+        hasItem("Warning: Your email usage just exceeded a configured threshold"))
+  }
+
+  @Test
+  def configurationShouldBeWellLoaded(server: GuiceJamesServer): Unit = {
+    val quotaProbe = server.getProbe(classOf[QuotaProbesImpl])
+    quotaProbe.setMaxStorage(quotaProbe.getQuotaRoot(MailboxPath.inbox(BOB)), QuotaSizeLimit.size(100 * 1000))
+
+    andreSendMailToBob(server)
+
+    // Bob receives a mail big enough to trigger a 10% configured threshold
+    awaitAtMostTwoMinutes.until(() => listMessageIds().size == 2)
+
+    andreSendMailToBob(server)
+
+    // Bob receives a mail big enough to trigger a 20% configured threshold
+    awaitAtMostTwoMinutes.until(() => listMessageIds().size == 4)
+    val ids: List[String] = listMessageIds().asScala.toList
+    val idString: String = concatMessageIds(ids)
+
+    val request =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core","urn:ietf:params:jmap:mail", "urn:ietf:params:jmap:submission"],
+         |  "methodCalls": [[
+         |     "Email/get",
+         |     {
+         |       "accountId": "$ACCOUNT_ID",
+         |       "ids": [$idString]
+         |     },
+         |     "c1"]]
+         |}""".stripMargin
+
+    `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post()
+    .`then`
+      .statusCode(SC_OK)
+      .log.ifValidationFails()
+      .body("methodResponses[0][1].list.preview",
```

Comment:
Why duplicate check `"methodResponses[0][1].list.preview"` here?
which one is correct?

Reply:
Both... it checks items in both previews (not sure can predict the return order?), checking that in the list we got both messages

Reply:
That's more or less what the draft test is doing too

Reply:
ah, understood